lib/src/main/java/com/shopify/checkoutsheetkit/ShopifyCheckoutSheetKit.kt

```diff
@@ -97,15 +98,16 @@ public object ShopifyCheckoutSheetKit {
      * @param context The context the checkout is being presented from
      * @param checkoutEventProcessor provides callbacks to allow clients to listen for and respond to checkout lifecycle events such as
      * (failure, completion, cancellation, external link clicks).
+     * @return the Dialog presented
      */
     @JvmStatic
     public fun <T: DefaultCheckoutEventProcessor> present(
         checkoutUrl: String,
         context: ComponentActivity,
         checkoutEventProcessor: T
-    ) {
+    ): Dialog? {
```

Comment:
I think this makes sense, but can I ask a little bit more about your use-case?

One concern with returning the Dialog, is that clients could potentially call Dialog methods (e.g. `setOnCancelListener()`, `setOnDismissListener()`) and override necessary cleanup.

What do you think about returning an interface that just allows dismissing the Dialog. Maybe something like the below (I'm thinking a `FunctionalInterface` for Java interoperability):

```kotlin
   @JvmStatic
    public fun <T: DefaultCheckoutEventProcessor> present(
        checkoutUrl: String,
        context: ComponentActivity,
        checkoutEventProcessor: T
    ): CheckoutSheetKitDialog? {
        if (context.isDestroyed || context.isFinishing) {
            return null
        }
        val dialog = CheckoutDialog(checkoutUrl, checkoutEventProcessor, context)
        context.lifecycle.addObserver(object: DefaultLifecycleObserver {
            override fun onDestroy(owner: LifecycleOwner) {
                dialog.dismiss()
                super.onDestroy(owner)
            }
        })
        dialog.start(context)
        return CheckoutSheetKitDialog { dialog.dismiss() }
    }
}

@FunctionalInterface
public fun interface CheckoutSheetKitDialog {
    public fun dismiss()
}
```